### DISCUSSION TITLE
Is possible to implement a callback after ApkUpdater.install ?

### DISCUSSION THREAD

**Discussion by CBR900cc** [kolbasa/cordova-plugin-apkupdater#48] [Ideas]
Hi @kolbasa , I love your plugin, but I need a little implementation, after calling ApkUpdater.install(), i need to know if the user accept or decline update request prompt

can you do this modification?

really thank you!

**Comment by kolbasa** [comment]
I did a little research. It seems that this is indeed possible.
I'll implement it this weekend. :+1:

**Reply by CBR900cc** [reply]
great!!! really thank you!!

**Comment by kolbasa** [comment]
I have now pushed some changes to the branch `install-results`.
Unfortunately, it's a bit more complicated than I thought.

There are two methods on this branch: `install` and `installDebug`.
Both have their pros and cons.

The `install` method is the one you already know.
However, I have modified it a bit. The callback waits for the user to return to the app.
If the user returns to the app, then it is actually safe to assume that they have cancelled the update.
Maybe this is already sufficient for your use case.

The `installDebug` method is new and actually checks if the installation was aborted or failed.
It even provides the error code if the APK could not be installed for some reason.
Stupidly, the method comes with a big drawback. 
The app is successfully updated, but the user does not get a dialog to reopen the app. He has to open the app on his own afterwards. I'm not sure if this can be fixed.
For the installation of other 3rd party apps, however, the method is great, there it works.

I'll take a break for now. I'll look again in the next few days to see if there's a better solution.

**Reply by CBR900cc** [reply]
very good news!! I'm really happy to know you're mutch fast on make modifications requested from users, thank you for your job!

I try new version asap
